domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidChildConstraintValidator.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.cmc.domain.constraints;
+
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Child;
+
+import java.util.Optional;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.mayNotBeNullError;
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.mayNotBeProvidedError;
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.setValidationErrors;
+import static uk.gov.hmcts.cmc.domain.models.statementofmeans.Child.AgeGroupType.BETWEEN_11_AND_15;
+import static uk.gov.hmcts.cmc.domain.models.statementofmeans.Child.AgeGroupType.BETWEEN_16_AND_19;
+import static uk.gov.hmcts.cmc.domain.models.statementofmeans.Child.AgeGroupType.UNDER_11;
+
+public class ValidChildConstraintValidator implements ConstraintValidator<ValidChild, Child> {
+
+    static class Fields {
+        static final String NO_OF_CHILDREN_LIVING_WITH_YOU = "numberOfChildrenLivingWithYou";
+        static final String NUMBER_OF_CHILDREN = "numberOfChildren";
+    }
+
+    @Override
+    public boolean isValid(Child child, ConstraintValidatorContext context) {
+        if (child == null) {
+            return true;
+        }
+
+        switch (child.getAgeGroupType()) {
+            case UNDER_11:
+            case BETWEEN_11_AND_15:
+                return validateYoungChild(child, context);
+            default:
+                return validateOlderChild(child, context);
+        }
+    }
+
+    private boolean validateOlderChild(Child child, ConstraintValidatorContext context) {
+        boolean valid = true;
+        String type = BETWEEN_16_AND_19.toString();
+
+        Optional<Integer> numberOfChildrenLivingWithYou = child.getNumberOfChildrenLivingWithYou();
+        int valueNoOfChildrenLivingWithYou = numberOfChildrenLivingWithYou.orElse(0);
+
+        if (!numberOfChildrenLivingWithYou.isPresent()) {
```

Comment:
I think it would be cleaner if we had:

```if (!child.getNumberOfChildrenLivingWithYou())```

And then below we would declare:

```int numberOfChildrenLivingWithYou = child.getNumberOfChildrenLivingWithYou().orElse(0);```

Reply:
ok